app/controllers/ingredients_controller.rb

```diff
@@ -0,0 +1,71 @@
+class IngredientsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_ingredient, only: %i[show edit update destroy]
+
+  def index
+    @ingredients = current_user.ingredients.all
+  end
+
+  def show; end
+
+  def new
+    @ingredient = current_user.ingredients.build
+    @recipe = current_user.recipes.find(params[:recipe_id])
+    @foods = Food.all
+  end
+
+  def create
+    @ingredient = current_user.ingredients.build(ingredient_params)
+    @recipe = current_user.recipes.find(params[:recipe_id])
+    respond_to do |format|
+      if @ingredient.save
+        format.html { redirect_to recipe_path(@recipe), notice: 'Ingredient saved successfully' }
+        format.json { render :show, status: :created, location: @ingredient }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @ingredient.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def edit
+    @ingredient = current_user.ingredients.find(params[:id])
+    @recipe = current_user.recipe
+  end
+
+  def update
+    @ingredient = Ingredient.find(params[:id])
+    respond_to do |format|
+      if @ingredient.update(ingredient_params)
+        format.html { redirect_to ingredient_url(@ingredient), notice: 'Ingredient was successfully updated.' }
+        format.json { render :show, status: :ok, location: @ingredient }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @ingredient.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
```

Comment:
- [ ] Please take a closer look at this wireframe and make the necesary update to your design, add ingredient button should be on the right and shopping list on the left, and make sure to widen your table to fit the screen for UI
EXPECTED

<img width="720" alt="1" src="https://github.com/AbbasSarwar/Recipe-app/assets/77843050/9e3a010b-7778-4db4-b671-794d107c4d1d">

YOURS
<img width="1215" alt="2" src="https://github.com/AbbasSarwar/Recipe-app/assets/77843050/ae2f58fb-dd64-41f1-ba60-886825147ac1">

Reply:
Done as advised.